src/config/agoric/agoric.tsx

```diff
@@ -61,33 +125,188 @@ const Agoric = () => {
     if (!walletAddress) {
       toast.error("Wallet not connected.", { autoClose: 3000 });
       throw new Error("wallet not connected");
     }
-    if (!isValidBundle(vals.bundle)) {
-      toast.error("Invalid bundle format.", { autoClose: 3000 });
-      throw new Error("Invalid bundle.");
-    }
-    const { compressedBundle, uncompressedSize } = await compressBundle(
-      JSON.parse(vals.bundle),
-    );
-    const proposalMsg = makeInstallBundleMsg({
-      compressedBundle,
-      uncompressedSize,
-      submitter: walletAddress,
-    });
-    try {
-      const txResponse = await signAndBroadcast(proposalMsg, "bundle");
-      if (txResponse) {
-        const { endoZipBase64Sha512 } = JSON.parse(vals.bundle);
-        await watchBundle(endoZipBase64Sha512, txResponse);
-        bundleFormRef.current?.reset();
+
+    const bundleObject = validateBundle(bundleString);
+    const { endoZipBase64Sha512 } = bundleObject;
+    const uncompressedBundleBytes = textEncoder.encode(bundleString);
+    const compressedBundleBytes = await gzip(uncompressedBundleBytes);
+    const compressedSize = compressedBundleBytes.byteLength;
+    const uncompressedSize = uncompressedBundleBytes.byteLength;
+
+    let blockHeight: number | undefined;
+
+    if (bundleString.length <= chunkSizeLimit) {
+      toast.info(
+        `Submitting bundle in one transaction, ${bytesFormat.format(
+          uncompressedSize,
+        )} uncompressed, ${bytesFormat.format(compressedSize)} (${(
+          (compressedSize / uncompressedSize) *
+          100
+        ).toFixed(0)}) compressed`,
+      );
+      const proposalMsg = makeInstallBundleMsg({
+        compressedBundle: compressedBundleBytes,
+        uncompressedSize: String(uncompressedSize),
+        submitter: walletAddress,
+      });
+      try {
+        const txResponse = await signAndBroadcast(proposalMsg, "bundle");
+        if (!txResponse) {
+          toast.error(
+            `No response for bundle submission transaction for bundle ${endoZipBase64Sha512}`,
+          );
+          return;
+        }
+        blockHeight = txResponse.height;
+      } catch (error) {
+        toast.error(
+          // @ts-expect-error it will be an Error, but null?.message would be fine anyway.
+          `Transaction failed to submit bundle to chain: ${error?.message}`,
+        );
+        return;
+      }
+    } else {
+      const { chunks, manifest } = await chunkBundle(
+        compressedBundleBytes,
+        chunkSizeLimit,
+      );
+
+      toast.info(
+        `Submitting bundle in ${
+          chunks.length + 1
+        } transactions (1 manifest and ${chunks.length} chunk${s(
+          chunks,
+        )}), ${bytesFormat.format(
+          uncompressedSize,
+        )} uncompressed, ${bytesFormat.format(
+          compressedSize,
+        )} (${percentFormat.format(
+          100 - (compressedSize / uncompressedSize) * 100,
+        )}%) compressed`,
```

Comment:
`style: "percent"` already scales 1 to 100 and adds the percent symbol.
```suggestion
        )} (${percentFormat.format(
          1 - compressedSize / uncompressedSize,
        )}) compressed`,
```
(but note that this is moot if you take my other suggestions)